indico/modules/categories/controllers/base.py

```diff
@@ -52,6 +62,39 @@ def _checkProtection(self):
                            .format(self.category.no_access_contact))
             raise Forbidden(' '.join(msg))
 
+    @staticmethod
+    def format_event_date(event):
+        day_month = "dd MMM"
+        if event.start_dt.year != event.end_dt.year:
+            return "{} - {}".format(format_date(event.start_dt), format_date(event.end_dt))
+        elif (event.start_dt.month != event.end_dt.month) or (event.start_dt.day != event.end_dt.day):
+            return "{} - {}".format(format_date(event.start_dt, day_month), format_date(event.end_dt, day_month))
+        else:
+            return format_date(event.start_dt, day_month)
+
+    def group_by_month(self, events):
+        def format_month(dt):
+            return format_date(dt, format='MMMM Y')
+        formatted_now = format_month(self.now)
+        months = []
```

Comment:
Wha about using an OrderedDict here instead of the `current_month` that gets updated whenever a new month starts? There's no ordered version of defaultdict but a `month = months.setdefault(.., {..})` call would do the job too